src/SIMD.jl

```diff
@@ -986,7 +1013,11 @@ for op in (:+, :-, :*, :/, :^, :copysign, :max, :min, :rem)
             llvmwrap(Val{$(QuoteNode(op))}, v1, v2)
     end
 end
-@inline Base. ^(v1::Vec{N,T}, x2::Integer) where {N,T<:FloatingTypes} =
+# Using `IntegerTypes` here so that this definition "wins" against
+# `^(::ScalarTypes, v2::Vec)`.
+@inline Base.:^(v1::Vec{N,T}, x2::IntegerTypes) where {N,T<:FloatingTypes} =
+    llvmwrap(Val{:powi}, v1, Int(x2))
+@inline Base.:^(v1::Vec{N,T}, x2::Integer) where {N,T<:FloatingTypes} =
```

Comment:
I just thought it's better since `Base.:^` is more explicit.  FYI, it looks like `julia` code base prefer `Base.:^`:

```console
$ git grep 'Base\. ^'
$ git grep 'Base\.:^'
base/compiler/ssair/show.jl:^(s::String, i::Int) = Base.:^(s, i)
base/mathconstants.jl:    Base.:^(::Irrational{:ℯ}, x::T) = exp(x)
doc/src/base/math.md:Base.:^(::Number, ::Number)
doc/src/base/strings.md:Base.:^(::AbstractString, ::Integer)
stdlib/LinearAlgebra/docs/src/index.md:Base.:^(::AbstractMatrix, ::Number)
stdlib/LinearAlgebra/docs/src/index.md:Base.:^(::Number, ::AbstractMatrix)
stdlib/LinearAlgebra/src/dense.jl:Base.:^(b::Number, A::AbstractMatrix) = exp!(log(b)*A)
stdlib/LinearAlgebra/src/dense.jl:Base.:^(::Irrational{:ℯ}, A::AbstractMatrix) = exp(A)
test/math.jl:Base.:^(x::Number, y::Float22716) = x^(y.x)
```

But I don't mind using `Base. ^`.  Let me know if I need to switch to it.

Reply:
I haven't seen this syntax before. If it works then it's fine. I'm mostly worried about consistency of style in the source code; people shouldn't wonder why there is `^` in one and `:^` in other places. If you prefer `:^`, then I'd prefer a pull request that changes this everywhere.

Reply:
It looks like this is the only place `Base.:^` is used but there are other similar cases like `Base. %`; I replaced them all.